src/pattern-library/components/patterns/input/CheckboxPage.js

```diff
@@ -0,0 +1,194 @@
+          </Library.Demo>
+        </Library.Example>
+      </Library.Pattern>
+
+      <Library.Pattern title="Props">
+        <Library.Example title="checked">
+          <p>
+            When the <code>checked</code> prop is present, <code>Checkbox</code>{' '}
+            will behave as a controlled component.
+          </p>
+          <Next.Code
+            content={`const [controlledChecked, setControlledChecked] = useState(false);
+
+const handleControlledChange = e => {
+  setControlledChecked(e.target.checked);
+};
+
+<Checkbox
+  checked={controlledChecked}
+  onChange={handleControlledChange}
+>
+  Controlled checkbox
+</Checkbox>
+`}
+            size="sm"
+            title="Controlled Checkbox usage example"
+          />
+          <Library.Demo title="Controlled Checkbox" withSource>
+            <Checkbox
+              checked={controlledChecked}
+              onChange={handleControlledChange}
+            >
+              Controlled checkbox
+            </Checkbox>
+          </Library.Demo>
+
+          <Library.Demo
+            title="Don't forget to handle events for controlled Checkboxes"
+            withSource
+          >
+            <Checkbox checked={false}>
+              <div className="text-red-error">
+                This checkbox will not update when clicked
+              </div>
+            </Checkbox>
+          </Library.Demo>
+        </Library.Example>
+
+        <Library.Example title="defaultChecked">
+          <p>
+            For uncontrolled <code>Checkbox</code>es,{' '}
+            <code>defaultChecked</code> sets initial checked state.
+          </p>
+          <Library.Demo
+            title="Uncontrolled Checkbox with defaultChecked"
+            withSource
+          >
+            <Checkbox defaultChecked>Default checked</Checkbox>
+          </Library.Demo>
+        </Library.Example>
+
+        <Library.Example title="disabled">
+          <p>
+            <code>disabled</code> styling reduces the opacity of the component.
+          </p>
+          <Library.Demo withSource>
+            <Checkbox>Enabled</Checkbox>
+            <Checkbox disabled>Disabled</Checkbox>
+          </Library.Demo>
+        </Library.Example>
+
+        <Library.Example title="icon and checkedIcon">
+          <p>
+            Custom icons may be used for styling checkboxes. Use the{' '}
+            <code>icon</code> and <code>checkedIcon</code> props to set custom{' '}
+            <code>IconComponent</code>s to use for the unchecked and checked
+            checkbox styling, respectively.
+          </p>
+          <Library.Demo withSource>
+            <Checkbox icon={HideIcon} checkedIcon={ShowIcon} defaultChecked />
+          </Library.Demo>
+        </Library.Example>
+
+        <Library.Example title="onChange">
+          <p>
+            Any provided <code>onChange</code> function is invoked with a{' '}
+            <code>TargetedEvent</code> when there is a <code>change</code> event
```

Comment:
There is no `TargetedEvent` interface/class in the web platform. All `Event` objects have a target, whether they are delivered to a DOM element or a global object (eg. document, window). The `change` event just receives a plain `Event` instance. I think the docs could just reference MDN here (eg. "onChange adds a handler for the ["change" event](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event) on the `<input>`").

In general, I think we can keep the docs succinct by assuming knowledge of how things work in HTML and deferring to MDN where a prop or other feature is a wrapper for an HTML attribute, event etc.